fabric/src/lib/java/dev/engine_room/flywheel/lib/model/baked/MeshEmitter.java

```diff
@@ -37,22 +41,21 @@ public BufferBuilder getBuffer(boolean shade) {
 	}
 
 	void prepareForGeometry(boolean shade) {
-		if (!bufferBuilder.building()) {
-			bufferBuilder.begin(VertexFormat.Mode.QUADS, DefaultVertexFormat.BLOCK);
-		} else if (shade != currentShade) {
+		bufferBuilder = new BufferBuilder(byteBufferBuilder, VertexFormat.Mode.QUADS, DefaultVertexFormat.BLOCK);
```

Comment:
I think this needs to move back into the if statement.

Placing a hopper minecart:
```
Caused by: java.lang.IllegalStateException: Not building!
	at com.mojang.blaze3d.vertex.BufferBuilder.ensureBuilding(BufferBuilder.java:76) ~[minecraft-merged-b25f1a0b75-1.21.1-loom.mappings.1_21_1.layered+hash.1839147940-v2.jar:?]
	at com.mojang.blaze3d.vertex.BufferBuilder.beginVertex(BufferBuilder.java:95) ~[minecraft-merged-b25f1a0b75-1.21.1-loom.mappings.1_21_1.layered+hash.1839147940-v2.jar:?]
	at com.mojang.blaze3d.vertex.BufferBuilder.addVertex(BufferBuilder.java:156) ~[minecraft-merged-b25f1a0b75-1.21.1-loom.mappings.1_21_1.layered+hash.1839147940-v2.jar:?]
	at dev.engine_room.flywheel.lib.model.baked.UniversalMeshEmitter.addVertex(UniversalMeshEmitter.java:60) ~[lib/:?]
```